mockk/src/main/kotlin/io/mockk/impl/annotations/JvmMockInitializer.kt

```diff
@@ -67,9 +78,113 @@ class JvmMockInitializer(val gateway: MockKGateway) : MockKGateway.MockInitializ
         }
     }
 
+    private fun resolveInstance(target: Any, property: KProperty1<Any, Any>): Any? {
+
+        var obj: Any?
+        val objCls = property.returnType.classifier as? KClass<*>
+
+        if (objCls != null) {
+            obj = objCls.objectInstance
+
+            if (obj == null) {
+                if (objCls.constructors.isEmpty() && objCls.primaryConstructor == null) {
+                    throw MockKException("No constructors found to instantiate @InjectMockK field ${property.name}")
+                } else {
+
+                    obj = resolveByConstructorInjection(target, objCls)
+
+                    if (obj == null) {
+                        try {
+                            val noArgCtor = objCls.constructors.first {
+                                it.parameters.isEmpty()
+                            }
+
+                            obj = noArgCtor.call()
+                            resolveByPropertyInjection(target, obj)
+                            (property as KMutableProperty1<Any, Any?>).set(target, obj)
+                        } catch (ex: NoSuchElementException) {
+                            throw MockKException("No suitable constructors found to instantiate @InjectMockK field ${property.name}")
+                        }
+                    } else {
+                        (property as KMutableProperty1<Any, Any?>).set(target, obj)
+                    }
+                }
+            } else {
+                throw MockKException("We could not inject mocks inside of @InjectMockK field ${property.name} because " +
```

Comment:
Probably it is not what we want to achieve. As it is possible that user has some code that relies on assigned values. Asking to change because of testing is what we want to avoid actually. Just overwrite values if needed. 

I would even try inject to `vals` via some dirty hack. This would add powerfulness.